R/compare.R

```diff
@@ -101,12 +146,54 @@ compare <- function(x, y, ...,
     ignore_attr = ignore_attr,
     ignore_encoding = ignore_encoding,
     ignore_formula_env = ignore_formula_env,
-    ignore_function_env = ignore_function_env
+    ignore_function_env = ignore_function_env,
+    ignore_name_order = ignore_name_order,
+    ignore_NULLs = ignore_NULLs,
+    ignore_private = ignore_private
   )
+
+  opts$user_specified <- c(intersect(names(match.call()),
+                                   names(opts)),
+                           "user_specified")
+
   out <- compare_structure(x, y, paths = c(x_arg, y_arg), opts = opts)
   new_compare(out, max_diffs)
 }
 
+sort_named_parts <- function(x) {
+  nx <- names(x)
+  if (!is.null(nx)) {
+    nonempty <- nchar(nx) > 0
+    ox <- order(nx[nonempty])
+    x[nonempty] <- x[nonempty][ox]
+    names(x)[nonempty] <- nx[nonempty][ox]
+  }
+  x
+}
+
+zap_nulls <- function(x) {
```

Comment:
FWIW our standard for this is:

```R
compact <- function(x) {
  is_null <- vapply(x, is.null, logical(1))
  x[!is_null]
}
```

And I'd put that in `utils.R`

Reply:
Done